staging/src/k8s.io/api/scheduling/v1alpha1/types.go

```diff
@@ -72,3 +72,106 @@ type PriorityClassList struct {
 	// items is the list of PriorityClasses
 	Items []PriorityClass `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Workload allows for expressing scheduling constraints that should be used
+// when managing lifecycle of workloads from scheduling perspective,
+// including scheduling, preemption, eviction and other phases.
+type Workload struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	//
+	// +optional
```

Comment:
Here's a place where declarative validation does well -- this doesn't say anything about the format of the name.  I can read validation code and find it, but API docs don't say anything.

Reply:
It's not common to say the name format here, even though we should.  Please do.